wrapper/src/main/java/software/amazon/jdbc/plugin/iam/ElastiCacheCredentialsProvider.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.plugin.iam;
+
+import io.lettuce.core.RedisCredentials;
+import io.lettuce.core.RedisCredentialsProvider;
+import reactor.core.publisher.Mono;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+import software.amazon.jdbc.PropertyDefinition;
+import software.amazon.jdbc.authentication.AwsCredentialsManager;
+import software.amazon.jdbc.util.Suppliers;
+
+public class ElastiCacheCredentialsProvider implements RedisCredentialsProvider {
+  private static final long TOKEN_EXPIRY_SECONDS = 900; // 15 minutes
+
+  private final String username;
+  private final String hostname;
+  private final int port;
+  private final String cacheName;
+  private final String region;
+  private final AwsCredentialsProvider awsCredentialsProvider;
+  private final Supplier<String> tokenSupplier;
+
+  public ElastiCacheCredentialsProvider(
+      String username,
+      String hostname,
+      int port,
+      String cacheName,
+      String region,
+      String awsProfile) {
+    this.username = username;
+    this.hostname = hostname;
+    this.port = port;
+    this.cacheName = cacheName;
+    this.region = region;
+    this.awsCredentialsProvider = createCredentialsProvider(awsProfile);
+    this.tokenSupplier = Suppliers.memoizeWithExpiration(
+        this::generateToken, TOKEN_EXPIRY_SECONDS, TimeUnit.SECONDS);
+  }
+
+  private AwsCredentialsProvider createCredentialsProvider(String awsProfile) {
+    if (awsProfile != null) {
+      Properties properties = new Properties();
+      properties.setProperty(PropertyDefinition.AWS_PROFILE.name, awsProfile);
```

Comment:
nitpick: maybe you can store `properties` containing just aws profile inside `CacheConnection.java` to avoid re-creating this Properties each time?